api/v1alpha1/volumereplication_types.go

```diff
@@ -20,20 +20,77 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// ImageState represents the replication operations to be performed on the image
+type ImageState string
+
+const (
+	// Primary ImageState enables mirroring and promotes the image to primary
+	Primary ImageState = "primary"
+
+	// Secondary ImageState demotes the image to secondary and resyncs the image if out of sync
+	Secondary ImageState = "secondary"
+
+	// Resync option resyncs the image
+	Resync ImageState = "resync"
```

Comment:
only `resync` action will be performed.

Reply:
I am not sure this answers the question. Adding my thoughts here for the use-case, it does not prevent this PR from proceeding though.

- If Desired state is resync, the actual state should be secondary
- If Desired state is resync, and actual state is primary it would report an error in a specific status, as backend replication may still be happening (hence a specific status field or something)

Reply:
yes it can be implemented in the controller

* We can check the image replication state first
* If it is secondary we can resync
* If it's primary we can report an error.

It's the user's responsibility to keep the actual state in secondary before resyncing.